### PR TITLE
Fixed reporting locked mock failure to jtreg logs (Issue #3)

### PR DESCRIPTION
**Pull request by NikolSkvarilova**

### CODE REVIEW

File: utils/core/base_xtest.py [hunk 1]
```diff
@@ -22,6 +22,10 @@ def defaultMain(argv, runDocs, runTests):
 
 
 class BaseTest(BaseTestRunner):
+    def __init__(self):
```
Comment (judovana) [inline utils/core/base_xtest.py:25]: I'm afraid this BaseTest extension , however excelent idea, should be in separate commit.

File: utils/mock/mock_executor.py [hunk 2]
```diff
@@ -133,8 +133,9 @@ def _rollbackCommand(self, name):
     def _scrubLvmCommand(self):
         o, e, r = exxec.processToStringsWithResult(self.mainCommand() + ["--scrub", "all"])
         if r != 0:
-            raise utils.mock.mock_execution_exception.MockExecutionException("Build chroot is locked, please restart "
-                                                                             "the testsuite.")
+            failed = "Build chroot is locked, please restart the testsuite."
```
Comment (judovana) [inline utils/mock/mock_executor.py:136]: What exactrly should change the chage from rise to log?
Afaik the raise was correctly killing the suite in case that chroot was useless. This wil lead to thausands of false failures.
But  Los Andrlos will have last word.

Comment (NikolSkvarilova) [reply utils/mock/mock_executor.py:136]: The 'raise' part is still there (line 138), but the failure is logged before it kills the suite.

Comment (judovana) [reply utils/mock/mock_executor.py:136]: I see. Yup I saw the rise later.. So the change was only for the log?
What the ```tu.passed_or_failed(self, False, failed)``` will do in this case?
Please split the refactoring and this chage.

File: utils/core/base_xtest.py (hunk 1 above)
Comment (andrlos) [reply utils/core/base_xtest.py:25]: The reason of the refactoring was the main purpose of this PR -> i dont see it as a refactoring and a change, its just a bigger change, where neccesary part of it was the refactoring. In the future, there wont be a time when you will want to revert just to the refactoring in this commit as it would prove useless. And the fix would be nonfunctional without the refactoring. Plus the change should have been there a long time ago, but only now I spotted this problem. 
I really dont see the need for splitting this into two commits other then forcing our intern into practicing git skills.

File: utils/mock/mock_executor.py (hunk 2 above)
Comment (andrlos) [reply utils/mock/mock_executor.py:136]: @judovana https://github.com/andrlos/jdk-rpms-specification-framework/issues/3 <- is the issue that is solved by this PR

Comment (andrlos) [reply utils/mock/mock_executor.py:136]: also, this change has already been thoroughly tested and works smoothly.

File: utils/core/base_xtest.py (hunk 1 above)
Comment (judovana) [reply utils/core/base_xtest.py:25]: To have refactoring separate from logic changes is mandatory practice.  It allows to see the logic change clear, and is really really good habit to get.
If it is  - as you are saying - inseparable change then tehre is something wrong. It may be my lack of python knowlege. Can  you please elaborate more?

**Comment by judovana** [comment]
@NikolSkvarilova  TYVM for splitting it up
